### DISCUSSION TITLE
Set temporary file location?

### DISCUSSION THREAD

**Discussion by brandonjbegle** [spatie/browsershot#644] [General]
Is it possible to set the temporary file directory?

In Browsershot.php, lines 780-785, in the function createTemporaryHtmlFile(), a new TemporaryDirectory() is created in the system temp directory. However, this doesn't work when using Laravel Homestead if chromium is installed with snap, since snap prevents file access pretty much beyond the home directory. 

I see that there is a location() function on the TemporaryDirectory class, but I can't see any way to set it beyond hardcoding something like

 $this->temporaryHtmlDirectory = (new TemporaryDirectory())->location('/home/vagrant/tmp')->create();

 inside the actual package function.
 
Thanks,

Brandon

**Comment by ccharz** [comment]
Try the [setCustomTempPath function](https://github.com/spatie/browsershot/blob/c2a8888661058322f6d039985abda7022bfd2a29/src/Browsershot.php#L120)  - it solved the problem for me

For example

```php
Browsershot::html('<p></p>')->setCustomTempPath(storage_path())->pdf();
```

**Reply by anuzpandey** [reply]
I did the same but its still giving me the same error.
```
Error: net::ERR_FILE_NOT_FOUND 
```
But the file exists at the given directory.

**Reply by sammyaxe** [reply]
same issue, did you figure out?

**Reply by heloufir** [reply]
Having the same issue, did you find any solution? I am using Sail on MacOS
